Work with database

BorodaPT opened the pull request:
1. Добавлены .sql файлы:
 - для создаения схемы БД schema.sql
 - для первичного заполенния справочных таблиц data.sql
2. Для работы с БД добавлены:
 - UserDbStorage : работа с таблицами пользователей;
 - FilmDbStorage : работа с таблицами фильмов;
 - MpaImpl : работа с возрастными ограничениями фильмов;
 - GenreImpl : работа с жанрами;
3. Добавлены классы сущностей MPA, Genre, Friend;
4. Добавлено поле Film хранения списков жанров в виде объектов Genre;
5. Добавлено поле Film.mpa для хранение возраствного ограничения
6. Добавлены тесты : FilmDbControllerTest, UserDbControllerTest;
7. Отредактированы методы тестов FilmControllerTest;

BorodaPT committed:
refactor:
1. Добавлены .sql файлы:
 - для создаения схемы БД schema.sql
 - для первичного заполенния справочных таблиц data.sql
2. Для работы с БД добавлены:
 - UserDbStorage : работа с таблицами пользователей;
 - FilmDbStorage : работа с таблицами фильмов;
 - MpaImpl : работа с возрастными ограничениями фильмов;
 - GenreImpl : работа с жанрами;
3. Добавлены классы сущностей MPA, Genre, Friend;
4. Добавлено поле Film хранения списков жанров в виде объектов Genre;
5. Добавлено поле Film.mpa для хранение возраствного ограничения
6. Добавлены тесты : FilmDbControllerTest, UserDbControllerTest;
7. Отредактированы методы тестов FilmControllerTest;

BorodaPT committed:
code-style

BorodaPT committed:
code-style

BorodaPT committed:
code-style

BorodaPT committed:
refactor: измененя подготовка данных для тестов getPopular и setLikeAdd в FilmDbControllerTest

db-exp commented on src/main/resources/schema.sql:
```diff
@@ -0,0 +1,70 @@
+);
+
+CREATE TABLE IF NOT EXISTS users (
+	id INTEGER GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
```
в модели  long, а это bigint

db-exp commented on src/main/resources/schema.sql:
```diff
@@ -0,0 +1,70 @@
+
+CREATE TABLE IF NOT EXISTS users (
+	id INTEGER GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+	login varchar NOT NULL,
```
на всех varchar нужно указать размерность, но не 255, а согласно хранимым данным

db-exp commented on src/main/resources/schema.sql:
```diff
@@ -0,0 +1,70 @@
+
+CREATE TABLE IF NOT EXISTS genre (
+	id INTEGER GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+	name varchar NOT NULL
```
размерность

db-exp commented on src/main/resources/schema.sql:
```diff
@@ -0,0 +1,70 @@
+);
+
+CREATE TABLE IF NOT EXISTS friends (
+	user_id INTEGER  NOT NULL,
```
bigint

db-exp commented on src/main/resources/schema.sql:
```diff
@@ -0,0 +1,70 @@
+
+CREATE TABLE IF NOT EXISTS films (
+	id INTEGER GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+	name varchar NOT NULL,
```
на всех varchar нужно указать размерность, но не 255, а согласно хранимым данным

db-exp commented on src/test/java/ru/yandex/practicum/filmorate/controller/FilmDbControllerTest.java:
```diff
@@ -0,0 +1,131 @@
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
+public class FilmDbControllerTest {
+
+    private final FilmDbStorage filmStorage;
```
лучше явно добавить qualifier

db-exp commented on src/main/java/ru/yandex/practicum/model/Genre.java:
```diff
@@ -0,0 +1,45 @@
+
+@RequiredArgsConstructor
+@Data
+
```
лишняя строка

db-exp commented on src/main/java/ru/yandex/practicum/model/Genre.java:
```diff
@@ -0,0 +1,45 @@
+    private int id;
+
+    @Nullable
+    private String name;
```
не уверен, что жанр может быть пустым

db-exp commented on src/main/java/ru/yandex/practicum/model/Mpa.java:
```diff
@@ -0,0 +1,32 @@
+    private int id;
+
+    @Nullable
+    private String name;
```
не уверен, что может быть пустым

db-exp commented on src/main/java/ru/yandex/practicum/model/Mpa.java:
```diff
@@ -0,0 +1,32 @@
+
+@Getter
+@Setter
+
```
лишняя строка

db-exp commented on src/main/java/ru/yandex/practicum/model/Film.java:
```diff
@@ -11,16 +11,16 @@
 
 @Getter
 @Setter
 
```
лишняя строка

BorodaPT committed:
refactor:
1. исправлен тип полей ID таблиц на BIGINT связанных с полями классов типа long
2. Добавлены органичения на длину полей varchar
3. Удалены строки с лишним кодом
4. в UserDbControllerTest и FilmDbControllerTest добавлен @Qualifier (надеюсь правильно понял про "явное добавление")
